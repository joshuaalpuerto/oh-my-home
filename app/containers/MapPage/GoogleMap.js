import React from 'react'
import PropTypes from 'prop-types'
import { withScriptjs, withGoogleMap, GoogleMap, Marker } from 'react-google-maps'

export class MapComponent extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  static propTypes = {
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired,
    images: PropTypes.array.isRequired
  }
  render () {
    const { lat, lng } = this.props
    return (
      <GoogleMap
        defaultZoom={17}
        defaultCenter={{ lat, lng }}
      >
        <Marker position={{ lat, lng }} />
      </GoogleMap>
    )
  }
}

export default withScriptjs(withGoogleMap(MapComponent))
