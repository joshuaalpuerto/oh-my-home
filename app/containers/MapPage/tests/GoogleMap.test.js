/**
 * Test the UsersPage
 */

import React from 'react'
import { shallow } from 'enzyme'

import GoogleMap from '../GoogleMap'
import {
  MAP_KEY
} from '../constants'

const wrapper = (props = {}, enzyme = shallow) => enzyme(
  <GoogleMap {...props} />
)

describe('<GoogleMap />', () => {
  const minProps = {
    lat: 1.123,
    lng: 2.123,
    images: [],
    googleMapURL: `https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${MAP_KEY}`,
    loadingElement: (<div style={{ height: `100%` }} />),
    containerElement: (<div style={{ height: `80vh` }} />),
    mapElement: (<div style={{ height: `100%` }} />),
    center: { lat: 1.123, lng: 1.123 }
  }

  it('render without exploding', () => {
    const renderComponent = wrapper(minProps)
    expect(
      renderComponent.length
    ).toEqual(1)
  })
})
