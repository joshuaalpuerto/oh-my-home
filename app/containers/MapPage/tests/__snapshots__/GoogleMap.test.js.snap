// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleMap /> Testing Without DOM should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <GoogleMap
    center={
        Object {
            "lat": 1.123,
            "lng": 1.123,
          }
    }
    defaultZoom={17}
>
    <Marker
        onClick={[Function]}
        position={
            Object {
                "lat": 1.123,
                "lng": 2.123,
              }
        }
    />
</GoogleMap>,
  "nodes": Array [
    <GoogleMap
      center={
            Object {
                  "lat": 1.123,
                  "lng": 1.123,
                }
      }
      defaultZoom={17}
>
      <Marker
            onClick={[Function]}
            position={
                  Object {
                        "lat": 1.123,
                        "lng": 2.123,
                      }
            }
      />
</GoogleMap>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <MapComponent
        center={
                Object {
                        "lat": 1.123,
                        "lng": 1.123,
                      }
        }
        containerElement={
                <div
                        style={
                                Object {
                                        "height": "80vh",
                                      }
                        }
                />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
        images={
                Array [
                        "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag",
                      ]
        }
        lat={1.123}
        lng={2.123}
        loadingElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
        mapElement={
                <div
                        style={
                                Object {
                                        "height": "100%",
                                      }
                        }
                />
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": MapComponent {
        "_infoToggle": [Function],
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "center": Object {
            "lat": 1.123,
            "lng": 1.123,
          },
          "containerElement": <div
            style={
                        Object {
                                    "height": "80vh",
                                  }
            }
/>,
          "googleMapURL": "https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag",
          "images": Array [
            "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag",
          ],
          "lat": 1.123,
          "lng": 2.123,
          "loadingElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
          "mapElement": <div
            style={
                        Object {
                                    "height": "100%",
                                  }
            }
/>,
        },
        "refs": Object {},
        "state": Object {
          "open": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <GoogleMap
          center={
                    Object {
                              "lat": 1.123,
                              "lng": 1.123,
                            }
          }
          defaultZoom={17}
>
          <Marker
                    onClick={[Function]}
                    position={
                              Object {
                                        "lat": 1.123,
                                        "lng": 2.123,
                                      }
                    }
          >
                    <InfoWindow
                              onCloseClick={[Function]}
                    >
                              <GoogleMap__ImageContainer>
                                        <Img
                                                  alt="https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
                                                  src="https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
                                        />
                              </GoogleMap__ImageContainer>
                    </InfoWindow>
          </Marker>
</GoogleMap>,
        "_debugID": 2,
        "_renderedOutput": <GoogleMap
          center={
                    Object {
                              "lat": 1.123,
                              "lng": 1.123,
                            }
          }
          defaultZoom={17}
>
          <Marker
                    onClick={[Function]}
                    position={
                              Object {
                                        "lat": 1.123,
                                        "lng": 2.123,
                                      }
                    }
          >
                    <InfoWindow
                              onCloseClick={[Function]}
                    >
                              <GoogleMap__ImageContainer>
                                        <Img
                                                  alt="https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
                                                  src="https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
                                        />
                              </GoogleMap__ImageContainer>
                    </InfoWindow>
          </Marker>
</GoogleMap>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MapComponent
    center={
        Object {
            "lat": 1.123,
            "lng": 1.123,
          }
    }
    containerElement={
        <div
            style={
                Object {
                    "height": "80vh",
                  }
            }
        />
    }
    googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag"
    images={
        Array [
            "https://maps.googleapis.com/maps/api/place/photo?maxwidth=400&photoreference=1234567&key=AIzaSyDDs-19AX317p7-U-ZNGv19NQtC1JO29Ag",
          ]
    }
    lat={1.123}
    lng={2.123}
    loadingElement={
        <div
            style={
                Object {
                    "height": "100%",
                  }
            }
        />
    }
    mapElement={
        <div
            style={
                Object {
                    "height": "100%",
                  }
            }
        />
    }
/>,
}
`;
